bears/xml/XMLBear.py

```diff
@@ -1,5 +1,26 @@
+import os
+import requests
+
 from coalib.bearlib.abstractions.Lint import Lint
 from coalib.bears.LocalBear import LocalBear
+from coalib.misc.Shell import escape_path_argument
+from coalib.settings.Setting import path
+
+
+def path_or_url(type1, type2):
+    '''
+    Checks if the argument is a ``url`` or a ``path``.
+
+    :param type1:       path
+    :param type2:       str
+    :return:            path or str
+    '''
+    def _path_or_url(dtdvalid):
+        if dtdvalid is not None and os.path.exists(dtdvalid):
+            return type1
```

Comment:
@AbdealiJK Still not able to find it in the log. I don't think it is even reaching that line. 
I have written this python script similar to the above program.https://gist.github.com/vivek425ster/2844337636d7e96391f6
It produces the following error:
```    f(10,s)
  File "annot.py", line 36, in f
    validate(f, locals())
  File "annot.py", line 5, in validate
    msg = 'Var: {0}\tValue: {1}\tTest: {2.__name__}	\tType: {3}'.format(var, value, test, type(test(value)))
  File "annot.py", line 25, in <lambda>
    return lambda dtdvalid: _path_or_url(dtdvalid)
  File "annot.py", line 22, in _path_or_url
    return x(dtdvalid)
  File "annot.py", line 11, in path
    return obj.__path__(*args, **kwargs)
AttributeError: 'str' object has no attribute '__path__'
```